lib/rules/semi.js

```diff
@@ -353,7 +355,7 @@ module.exports = {
                     report(node);
                 }
             } else {
-                const oneLinerBlock = (exceptOneLine && isLastInOneLinerBlock(node));
+                const oneLinerBlock = ((exceptOneLine || exceptOneLineClassBody) && isLastInOneLinerBlock(node));
```

Comment:
Per this condition, `"omitLastInOneLineClassBody": true` would also apply to blocks:

```js
/* eslint semi: ["error", "always", { "omitLastInOneLineClassBody": true}] */

{ foo; } // false positive

```

It might be clearer to add a new function `isLastInOneLinerClassBody`.

Reply:
@mdjermanovic Good catch. Fixed.